lib/spack/spack/platforms/cray.py

```diff
@@ -103,8 +109,15 @@ def setup_platform_environment(cls, pkg, env):
         """ Change the linker to default dynamic to be more
             similar to linux/standard linker behavior
         """
-        env.set('CRAYPE_LINK_TYPE', 'dynamic')
+        packages = spack.config.get_config("packages")
```

Comment:
This needs to be superseded by some sort of universal variant for `shared` or `link_type`.